README.md

```diff
@@ -70,10 +70,104 @@ You will see an complete code sample in MessageHelperTest.cs
          true,
          getNotify, culture, 1, 5)
       { FilesPath = files };
+      
+      // Attaching events
+      
+      fileHelper.Before_UploadChunksErrorRetry += (IEnumerable<int> chunks, string inputFile, Guid fileIdentifier) =>
+      {
+          /// This is the delegate associated to the event when chunks are in errors, before retry upload.
+          /// <param name="chunks">Chunks list</param>
+          /// <param name="inputFile">File path</param>
+          /// <param name="fileIdentifier">File ID</param>
+      };
+      
+      fileHelper.UploadFileStart_Event += (string eventFileName, long fileSize) =>
+      {
+          /// This event occur before the file start to be uploaded to the server.
+          var fileName = eventFileName;
+          var theFileSize = fileSize;
+      };
+
+      fileHelper.UploadFileEnd_Event += (string eventFileName, long fileSize) =>
+      {
+          /// This event occur when all the file chunks has finish
+      };
+     
+      fileHelper.UploadFinish_Event += () =>
+      {
+          /// The event call when a chuck finish upload
+      };
+
+      fileHelper.UploadFinishFiles_Event += (List<UploadFiles> uploadedFiles) =>
+      {
+          /// This event occur when all the files upload has finish
+          for (int i = 0; i < uploadedFiles.Count; i++)
+          {
+              UploadFiles upf = uploadedFiles[i]; // That is the model of files chunk when file are uploaded to the Secure Exchanges server
+              
+              string fileName = upf.RealFileName; // Real file name of the file
+              
+              int fileLength = upf.BytesLenght; // The file length (the size of the file). If you need the beauty filesize, you can use the GetTextSize
+
+              string fileHash = upf.Hash; // The Hex MD5 file hash
```

Comment:
The upf.Hash is now always SHA512. MD5 are no more supported by Secure Exchanges. Thant mean every property hash has been updated to reflect the SHA512 file hash.